addons/web/static/tests/views/list_tests.js

```diff
@@ -10238,6 +10238,100 @@ QUnit.module('Views', {
+                assert.ok(dateRegex.test(evalContext.today));
+                assert.strictEqual(evalContext.current_date, evalContext.today);
+                return evalContext;
+            },
+        });
+        /**
+         * Returns "1997-01-DD" with D holding current UTC date from patched date
+         * + (deltaDays) days.
+         * This is done to allow testing from any timezone since UTC values are
+         * calculated with the offset of the current browser.
+         */
+        function dateStringDelta(deltaDays) {
+            const d = new Date(Date.now() + 1000 * 60 * 60 * 24 * deltaDays);
+            return `1997-01-${String(d.getUTCDate()).padStart(2, '0')}`;
+        }
+
+        this.data.foo.records[0].date = dateStringDelta(-1);
+        this.data.foo.records[1].date = dateStringDelta(0);
+        this.data.foo.records[2].date = dateStringDelta(+1);
+
+        const list = await createView({
+            arch: `
+                <tree>
+                    <field name="date" decoration-danger="date > today"/>
+                </tree>`,
+            data: this.data,
+            model: 'foo',
+            View: ListView,
+        });
+
+        assert.containsOnce(list, ".text-danger");
+
+        list.destroy();
+        unpatchDate();
+        testUtils.mock.unpatch(BasicModel);
+    });
+
+    QUnit.test("Datetime in evaluation context works with datetime field", async function (assert) {
+        assert.expect(6);
+
+        const datetimeRegex = /^\d{4}\-\d{2}\-\d{2} \d{2}:\d{2}:\d{2}$/;
+        const unpatchDate = testUtils.mock.patchDate(1997, 0, 9, 12, 0, 0);
+        testUtils.mock.patch(BasicModel, {
+            _getEvalContext() {
+                const evalContext = this._super(...arguments);
+                assert.ok(datetimeRegex.test(evalContext.now));
+                return evalContext;
+            },
+        });
+        /**
+         * Returns "1997-01-DD HH:MM:00" with D, H and M holding current UTC values
+         * from patched date + (deltaMinutes) minutes.
+         * This is done to allow testing from any timezone since UTC values are
+         * calculated with the offset of the current browser.
+         */
+        function dateStringDelta(deltaMinutes) {
+            const d = new Date(Date.now() + 1000 * 60 * deltaMinutes);
+            return `1997-01-${
+                String(d.getUTCDate()).padStart(2, '0')
+            } ${
+                String(d.getUTCHours()).padStart(2, '0')
+            }:${
+                String(d.getUTCMinutes()).padStart(2, '0')
+            }:00`;
+        }
+        this.data.foo.records[0].datetime = dateStringDelta(-30);
+        this.data.foo.records[1].datetime = dateStringDelta(0);
+        this.data.foo.records[2].datetime = dateStringDelta(+30);
+
+        const list = await createView({
+            arch: `
+                <tree>
+                    <field name="datetime" decoration-danger="datetime > now"/>
```

Comment:
I find it slightly confusing to name the field `datetime` in the test, as it collides with the `datetime` object always present in the eval context. So `"datetime > now"` looks like it has two meanings.

Perhaps `datetime_field` would work better, as `"datetime_field > now"` doesn't create that confusion? Same thing with `"date_field > today"` above, even though there is no collision in this case - it would still feel more readable.

Unless of course the point of the test was to make sure field values _do override_ default context "built-ins" (but it doesn't look like it)